### PR TITLE
[#260] Fix: Favorite button doesn't update correctly on Home Screen

### LINKED ISSUE #260: Fix: Favorite button doesn't update correctly on Home Screen
## Issue

Favorite button doesn't update count on Home Screen, and after refresh the page the Favorite button's highlight state doesn't show correctly that I already favorite an article but it is not highlighted.

https://user-images.githubusercontent.com/70877098/143375927-9fce4b92-a805-4c22-b730-9ea41e775a45.MP4

## Expected

The favorite button status should be updated correctly per user's preferences.

## Steps to reproduce

1. Open the app with or without valid credentials.
2. Open global feeds page.
3. Click on favorite button of any article.
4. Observe result.
5. Pull to refresh the global feeds.
3. Observe the result.

### PR DESCRIPTION
**Pull request by markgravity**
Resolved #260

## What happened

Fix favorite button update incorrectly

## Insight

Please ignore the typo `favourite`, it will be renamed in #263

## Proof Of Work

https://user-images.githubusercontent.com/17875522/143379543-d14141fd-5c71-4516-8553-a5fabd9c392d.mov

### CODE REVIEW

File: NimbleMedium/Sources/Presentation/Modules/Feeds/ArticleRow/ArticleRowViewModel.swift [hunk 1]
```diff
@@ -93,33 +103,44 @@ extension ArticleRowViewModel: ArticleRowViewModelOutput {}
...
             .mapToVoid()
             .catchAndReturn(())
     }
 
-    private func updateToggleFavouriteArticle() -> Observable<Bool> {
+    private func updateToggleFavouriteArticle() -> Observable<(Bool, Int)> {
         guard let uiModel = $uiModel.value else { return .empty() }
-        updateFavouriteArticle(!uiModel.articleIsFavorited)
+        let isFavourite = uiModel.articleIsFavorited
+        let count = uiModel.articleFavoriteCount + 1 * (!isFavourite ? 1 : -1)
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/Feeds/ArticleRow/ArticleRowViewModel.swift:134]: I think we can remove `1 *` part. What do you think?

File: NimbleMedium/Sources/Presentation/Modules/ArticleDetail/ArticleDetailViewModel.swift [hunk 2]
```diff
@@ -220,31 +229,44 @@ extension ArticleDetailViewModel {
...
             .mapToVoid()
             .catchAndReturn(())
     }
 
-    private func updateToggleFavouriteArticle() -> Observable<Bool> {
+    private func updateToggleFavouriteArticle() -> Observable<(Bool, Int)> {
         guard let uiModel = $uiModel.value else { return .empty() }
-        updateFavouriteArticle(!uiModel.articleIsFavorited)
+        let isFavourite = uiModel.articleIsFavorited
+        let count = uiModel.articleFavoriteCount + 1 * (!isFavourite ? 1 : -1)
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Modules/ArticleDetail/ArticleDetailViewModel.swift:260]: Ditto here 🙏

File: NimbleMedium/Sources/Data/Repositories/Article/ArticleRepository.swift [hunk 3]
```diff
@@ -10,14 +10,9 @@ import RxSwift
 final class ArticleRepository: ArticleRepositoryProtocol {
 
     private let authenticatedNetworkAPI: AuthenticatedNetworkAPIProtocol
-    private let networkAPI: NetworkAPIProtocol
 
-    init(
-        authenticatedNetworkAPI: AuthenticatedNetworkAPIProtocol,
-        networkAPI: NetworkAPIProtocol
-    ) {
+    init(authenticatedNetworkAPI: AuthenticatedNetworkAPIProtocol) {
```
Comment (suho) [inline NimbleMedium/Sources/Data/Repositories/Article/ArticleRepository.swift:14]: Why don't you use `Resolver` here?

Comment (suho) [reply NimbleMedium/Sources/Data/Repositories/Article/ArticleRepository.swift:14]:
```swift
@Injtected let authenticatedNetworkAPI: AuthenticatedNetworkAPIProtocol
```

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

**Review by suho — APPROVED** [review]

**Pull request merged**